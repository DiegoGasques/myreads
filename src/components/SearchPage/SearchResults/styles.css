.list-item {
  padding: 20px;
}

.search-books-results {
  padding: 80px 10px 20px;
}
