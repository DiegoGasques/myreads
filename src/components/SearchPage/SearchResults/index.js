import React from "react";
import PropTypes from "prop-types";

import List from "../../List";
import BookCard from "../../BookCard";
import { AppContext } from "../../App";

import "./styles.css";

function SearchResults({ results }) {
  return (
    <div className="search-books-results">
      <AppContext.Consumer>
        {context => (
          <List>
            {results.map(b => {
              return (
                <li key={`${b.id}`} className="list-item">
                  <BookCard book={b} handleUpdate={context.updateBookStatus} />
                </li>
              );
            })}
          </List>
        )}
      </AppContext.Consumer>
    </div>
  );
}

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default SearchResults;
