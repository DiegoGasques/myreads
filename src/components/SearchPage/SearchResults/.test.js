import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SearchResults from "./index";

import { books } from "../../../testFixtures";

it("Should match SearchBar component snapshot", () => {
  const wrapper = shallow(<SearchResults results={books} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
