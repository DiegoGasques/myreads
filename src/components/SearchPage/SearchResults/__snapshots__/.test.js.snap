// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match SearchBar component snapshot 1`] = `
<div
  className="search-books-results"
>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</div>
`;
